### PR TITLE
k8s: add pod OOM test

### PR DESCRIPTION
**Pull request by liubin**
Fixes: #2853

Signed-off-by: bin liu <bin@hyper.sh>

### CODE REVIEW

File: integration/kubernetes/k8s-oom.bats [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	sleep_time=2
+
+	# Create pod
+	kubectl create -f "${pod_config_dir}/pod-oom.yaml"
+
+	# Check pod creation
+	kubectl wait --for=condition=Ready pod "$pod_name"
+
+	# Check if OOMKilled
+	cmd="kubectl get pods "$pod_name" -o yaml | yq r -  'status.containerStatuses[0].state.terminated.reason' | grep OOMKilled"
```
Comment (jodh-intel) [inline integration/kubernetes/k8s-oom.bats:28]: Nit: double space after `yq -r -`.

**Review by jodh-intel — APPROVED** [review]
Thanks @liubin.

lgtm

File: integration/kubernetes/k8s-oom.bats (hunk 1 above)
Comment (liubin) [reply integration/kubernetes/k8s-oom.bats]: Nice catch

**Comment by fidencio** [comment]
@liubin, please, feel free to send me a notification when this one is ready to be reviewed.

**Comment by liubin** [comment]
Hi @fidencio , it's now ready to review.

**Review by fidencio — APPROVED** [review]
lgtm!

**Pull request merged**